zephyr/CMakeLists.txt

```diff
@@ -330,13 +330,47 @@ endif()
 # NXP IMX8 platforms
 if (CONFIG_SOC_SERIES_NXP_IMX8)
 	zephyr_library_sources(
-		#${SOF_DRIVERS_PATH}/generic/dummy-dma.c
-		#${SOF_DRIVERS_PATH}/imx/sdma.c
-		#${SOF_DRIVERS_PATH}/imx/edma.c
-		#${SOF_DRIVERS_PATH}/imx/sai.c
-		#${SOF_DRIVERS_PATH}/imx/ipc.c
-		#${SOF_DRIVERS_PATH}/imx/esai.c
+		${SOF_DRIVERS_PATH}/generic/dummy-dma.c
+		${SOF_DRIVERS_PATH}/imx/edma.c
+		${SOF_DRIVERS_PATH}/imx/sai.c
+		${SOF_DRIVERS_PATH}/imx/ipc.c
+		${SOF_DRIVERS_PATH}/imx/esai.c
 	)
+
+	# Platform sources
+	zephyr_library_sources(
+		${SOF_PLATFORM_PATH}/imx8/platform.c
+		${SOF_PLATFORM_PATH}/imx8/lib/clk.c
+		${SOF_PLATFORM_PATH}/imx8/lib/dai.c
+		${SOF_PLATFORM_PATH}/imx8/lib/dma.c
+		${SOF_PLATFORM_PATH}/imx8/lib/memory.c
+	)
+
+	set(PLATFORM "imx8")
+endif()
```

Comment:
@iuliana-prodan we need to be aware that we have two platforms (imx8 and imx8x) although very similar. Not sure what PLATFORM stands for here.

Reply:
This part here includes all the files from sof/src/platform/imx8/ which are common for both imx8 and imx8x.
I've set PLATFORM as imx8 since this is the only one I've tested and focused my work on this.

When I'll check on imx8x will see if I need to modify this or not. Right now, I'm not sure.

Reply:
imx8 and imx8x have minor differences in the kconfig, really. They are extremely similar but not the same. Basically all the code is shared but addresses of hardware and shit differ.